llama_stack/providers/remote/inference/vllm/vllm.py

```diff
@@ -203,4 +203,14 @@ async def embeddings(
         model_id: str,
         contents: List[InterleavedTextMedia],
     ) -> EmbeddingsResponse:
-        raise NotImplementedError()
+        model = await self.model_store.get_model(model_id)
+
+        kwargs = {}
+        if model.metadata.get("embedding_dimensions"):
```

Comment:
is this optional? i.e., should you not assert that it always exists?

Reply:
fixed to be an assertion.